### PR TITLE
Choose variables

### PR DESCRIPTION
**Pull request by catdieval**
The user can select which variable to plot for the x-axis and the y-axis with dropdown menus.

### CODE REVIEW

File: Components/ConvertCSVToArray/ConvertCSVToArrayOfObjects.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    return row !== "";
+  });
+
+  const array = [];
+
+  let headerRow;
+  let headerObj;
+  const content = [];
+
+  rows3.forEach((row, idx) => {
```
Comment (NikRoe) [inline Components/ConvertCSVToArray/ConvertCSVToArrayOfObjects.js:29]: I find this forEach rather hard to follow due to the naming within.

Please take another look at it and name arguments, parameters and variables in a way that makes it clear what they stand for

File: pages/_app.js [hunk 2]
```diff
@@ -1,11 +1,100 @@
...
+
+  function handleYChange(event) {
+    const choice = event.target.value;
+    setYKey(choice);
+  }
+
+  function handleAssignVariables() {
+    if (vals.length > 0 && xKey != "" && yKey != "") {
+      var tempXArray = new Array(vals.length);
+      var tempYArray = new Array(vals.length);
```
Comment (NikRoe) [inline pages/_app.js:49]: why `var` and not `let`?

File: pages/_app.js [hunk 3]
```diff
@@ -1,11 +1,100 @@
...
+
+      // In tempXArray fill in with the values associated to the key in vals
+      // corresponding to the xKey value.
+      // In tempYArray fill in with the values associated to the key in vals
+      // corresponding to the yKey value.
+
+      for (let i = 0; i < vals.length; i++) {
+        for (let key in vals[i]) {
+          if (key == xKey) {
```
Comment (NikRoe) [inline pages/_app.js:58]:
```suggestion
          if (key === xKey) {
```

File: pages/_app.js [hunk 4]
```diff
@@ -1,11 +1,100 @@
...
+            tempXArray[i] = vals[i][key];
+          }
+          if (key == yKey) {
```
Comment (NikRoe) [inline pages/_app.js:61]:
```suggestion
          if (key === yKey) {
```

**Review by NikRoe — CHANGES_REQUESTED** [review]
Hi there,

nice start!

I noticed a few things that could be improved and left some suggestions.
Please take a look at those.

Keep it up 🙌

File: Components/ChooseVariables/index.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  onXChange,
+  onYChange,
+  xKey,
+  yKey,
+}) {
+  function handleSubmit(event) {
+    event.preventDefault();
+  }
+
+  if (xKey == yKey && xKey !== "") {
```
[the remark this answers is not in the document]
Comment (catdieval) [reply Components/ChooseVariables/index.js:19]: Done.

File: Components/ChooseVariables/index.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+              <br></br>
+              <StyledInputTypeSubmit
+                value="Next"
+                onClick={onAssignVariables}
+                disabled={completedXAndYKeys()}
+              />
+            </CenteredDiv>
+          </form>
+        </>
+      ) : null}
```
[the remark this answers is not in the document]
Comment (catdieval) [reply Components/ChooseVariables/index.js:58]: Done.

File: Components/ConvertCSVToArray/CheckIfValid.js [hunk 7]
```diff
@@ -0,0 +1,9 @@
+export default function CheckIfValid(data, { separator }) {
```
[the remark this answers is not in the document]
Comment (catdieval) [reply Components/ConvertCSVToArray/CheckIfValid.js:1]: Done.

File: Components/ConvertCSVToArray/ConvertCSVToArrayOfObjects.js [hunk 8]
```diff
@@ -0,0 +1,60 @@
+import { convertStringToNumber } from "convert-string-to-number";
+
+export default function ConvertCSVToArrayOfObjects(
```
[the remark this answers is not in the document]
Comment (catdieval) [reply Components/ConvertCSVToArray/ConvertCSVToArrayOfObjects.js:3]: Done.

File: Components/ConvertCSVToArray/CheckIfValid.js
[the remark this answers is not in the document]
Comment (catdieval) [reply Components/ConvertCSVToArray/CheckIfValid.js:1]: Done.

File: pages/_app.js (hunk 4 above)
Comment (catdieval) [reply pages/_app.js:61]: Done.

File: pages/_app.js (hunk 3 above)
Comment (catdieval) [reply pages/_app.js:58]: Done.

File: pages/_app.js (hunk 2 above)
Comment (catdieval) [reply pages/_app.js:49]: We have changed var to let.

File: Components/ListOfCharts/index.js [hunk 9]
```diff
@@ -3,22 +3,26 @@ import { Container } from "../Container";
...
+                  key={chart.name}
+                  icon={chart.icon}
+                  name={chart.name}
+                  onSelectChartType={onSelectChartType}
+                />
+              );
+            })}
+          </Container>
+        </>
+      ) : null}
```
[the remark this answers is not in the document]
Comment (catdieval) [reply Components/ListOfCharts/index.js:25]: Done.

File: Components/DropDownMenu/index.js [hunk 10]
```diff
@@ -0,0 +1,15 @@
+export default function DropDownMenu({ idString, onChange, arrayOfOptions }) {
+  return (
+    <>
+      <label htmlFor={idString}>{idString}:</label>
+      <select required name={idString} id={idString} onChange={onChange}>
+        <option value="">Select</option>
+        {arrayOfOptions.map((optionname) => (
+          <option key={optionname} value={optionname}>
+            {optionname}
+          </option>
```
[the remark this answers is not in the document]
Comment (catdieval) [reply Components/DropDownMenu/index.js:10]: Done.

File: Components/ConvertCSVToArray/ConvertCSVToArrayOfObjects.js (hunk 1 above)
Comment (catdieval) [reply Components/ConvertCSVToArray/ConvertCSVToArrayOfObjects.js]: We changed idx to index.
